### DISCUSSION TITLE
mysql2 gem error with ruby 2.7 on aws lambda

### DISCUSSION THREAD

**Discussion by davidplappert** [planetscale/discussion#117] [Bug Reports]
Mary & Liz asked me to open this issue. 

Liz - Can you please copy the error messages I sent you? I didn't save them when I sent them via Zoom to you.

**Comment by lizztheblizz** [comment]
Hey David! Yes, let me fill out some context here. 

Your environment is the following:
- Ruby 2.7 on lambda
- mysql2 (0.5.3)

The error you were receiving upon your gem trying to connect was as follows:
```
{
    "errorMessage": "SSL connection error: unknown error number",
    "errorType": "Function<ActiveRecord::ConnectionNotEstablished>",
    "stackTrace": [
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {}
    ]
}
```

The connection details were defined as follows:
```
  require 'active_record'
  require 'mysql2-lambda'
  require 'mysql2'
  ActiveRecord::Base.establish_connection(
    adapter: 'mysql2',
    database: 'xxx',
    username: 'xxx',
    host: 'xxx',
    password: 'xxx,
    sslverify: true,
    # ssl_mode: :verify_identity,
    # sslca: '/etc/ssl/cert.pem',
  )
```

**Reply by lizztheblizz** [reply]
Lambda gem thats involved appears to live here: https://github.com/customink/mysql2-lambda

**Reply by lizztheblizz** [reply]
Might be related to the MySQL Connector being used in this gem being quite old?
https://github.com/customink/mysql2-lambda/blob/e5748e875e679384bdea031b760b1646b0bf35dc/Dockerfile#L22

**Comment by jgreet** [comment]
I noticed https://github.com/customink/mysql2-lambda/issues/3, is that the version of the mysql2 gem you're using?

**Reply by davidplappert** [reply]
yes
